src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1203,7 +1203,7 @@ protected void setAttribute(final short tag, final Object value) {
 
     protected void setAttribute(final short tag, final Object value, final boolean isUnsignedArray) {
         if (value != null &&
-                !(value instanceof Byte || value instanceof Short || value instanceof Integer ||
+                !(value instanceof Byte || value instanceof Short || value instanceof Integer || value instanceof Long ||
```

Comment:
With this change to allow `Long` values directly in `SAMRecord` attributes, what protection do we have against a client setting an attribute outside the range of `uint32_t` that bams can handle? The htsjdk BAM codec will fail given a `Long` value that exceeds `BinaryCodec.MAX_UINT`.

Also, if you set such an attribute using `setAttribute()` and then call `getIntegerAttribute()` you'll get a `RuntimeException` (even though the attribute may be marked as of type i/I, like the `ia:i:4294967295` attribute in http://gatkforums.broadinstitute.org/discussion/comment/24038/#Comment_24038).

Can you comment on these concerns? I don't want to block this important fix, but I want to be sure we're not doing something here that will cause problems later on.